File: part3/mean-messageapp/backend/service/TokenStorage.js
class TokenStorage {
  constructor() {
    this.tokens = []; 
  }

  addToken(token) {
    this.tokens.push(token); 
  }

  removeToken(token) {
    const index = this.tokens.indexOf(token);
    if (index !== -1) {
      this.tokens.splice(index, 1); 
    }
  }

  findInTokens(token) {
    return this.tokens.includes(token); 
  }
}

module.exports = new TokenStorage();

File: part2/messageapp/src/main/java/com/yildirim/messageapp/TokenStorage.java
package com.yildirim.messageapp;

import org.apache.commons.lang3.ArrayUtils;

public class TokenStorage {
  private static String[] tokens;

  public static boolean findInTokens(String token) {
    return ArrayUtils.contains(tokens, token);
  }

  public static void addToken(String token) {
    tokens = ArrayUtils.add(tokens, token);
  }

  public static void removeToken(String token) {
    tokens = ArrayUtils.removeElement(tokens, token);
  }
}
